test/test_torch.py

```diff
@@ -3864,9 +3864,16 @@ def test_print(self):
                 str(obj)
 
             # test complex tensor
+            # complex tensor print uses two formatters, one for real values
```

Comment:
Meta question: these tests seem OK, but would it be possible to compare more directly with NumPy's printing? 

The only differences should be that NumPy writes "array" instead of "tensor" and will write the dtype at difference times since both PyTorch and NumPy only write a tensor's dtype if it's not equal to the default float scalar type. Right? Seems like this test could account for those differences.

Reply:
There are other differences between numpy array and torch tensor printing like extra 0s, when we choose to have sci_mode etc, which is prob why other dtypes currently don't compare torch out with numpy out